pkg/ccl/kvccl/kvtenantccl/tenant_upgrades.md

```diff
@@ -0,0 +1,83 @@
+# Upgrading Tenants
+
+What does it even mean to think about upgrading tenants?
+The upgrade protocol for KV nodes has become well-defined
+over the course of 21.1. It involves tight coordination between
+nodes to gain invariants about what features will be used and
+what migrations can and have been run. SQL pods are inherently
+much more ephemeral and less able to perform this coordination.
+Nevertheless, we need some way to migrate data in tenants and we
+need some claims about the features the tenants may end up using.
+
+* Tenants have their own system tables, including settings.
+* System tables and initial data get bootstrapped in CreateTenant.
+    * We should probably write out an initial version into the settings table
+      at that point.
+    * For new tenants what should we initialize this to?
+        * The best that the system tenant can do is its
+          active version.
+    * For pre-existing tenants in the 20.2->21.1 upgrade, cope
```

Comment:
That, or we can land a patch release that has a sqlmigration that inserts v20.2 into the table and make sure that CC deploys that throughout all of the free tier pods. They will likely need to roll out a patch release anyway in order to be able to do the upgrade, so might as well bake it in there avoid any exceptions in the actual code.